### DISCUSSION TITLE
Getting an error when creating the docker

### DISCUSSION THREAD

**Discussion by vcrenjith** [pkjmesra/PKScreener#46] [Q&A]
talib/_ta_lib.c:196:12: fatal error: longintrepr.h: No such file or directory
2.152         196 |   #include "longintrepr.h"
2.152             |            ^~~~~~~~~~~~~~~
2.152       compilation terminated.
2.152       error: command '/usr/bin/gcc' failed with exit code 1
2.152       [end of output]
2.152   
2.152   note: This error originates from a subprocess, and is likely not a problem with pip.
2.153   ERROR: Failed building wheel for ta-lib
2.153   Running setup.py clean for ta-lib
2.396 Failed to build ta-lib
2.396 ERROR: Could not build wheels for ta-lib, which is required to install pyproject.toml-based projects
2.432 
2.432 [notice] A new release of pip is available: 23.2.1 -> 23.3.2
2.432 [notice] To update, run: pip install --upgrade pip
------
Dockerfile:39
--------------------
  37 |     
  38 |     RUN rm -rf ta-lib ta-lib-0.4.0-src.tar.gz
  39 | >>> RUN pip3 install ta-lib==${PYTHON_TA_LIB_VERSION}
  40 |     
  41 |     CMD ["python3"]
--------------------
ERROR: failed to solve: process "/bin/sh -c pip3 install ta-lib==${PYTHON_TA_LIB_VERSION}" did not complete successfully: exit code: 1

**Comment by vcrenjith** [comment]
Getting above message when trying to deploy the docker container.

**Reply by pkjmesra** [reply]
@vcrenjith - Please note that if you're trying to build the image locally, you'd need to have ta-lib and its dependencies pre-installed. Please see https://github.com/TA-Lib/ta-lib-python?tab=readme-ov-file#windows for pre-requisites.

If you're building from source, you can also look at the workflow file https://github.com/pkjmesra/PKScreener/blob/main/.github/workflows/w1-workflow-build-matrix.yml for all supported operating systems and follow the steps therein for building ta-lib.

Have you tried just : 
`docker pull pkjmesra/pkscreener:latest` followed by
`docker run -it pkjmesra/pkscreener:latest` ?

**Comment by vcrenjith** [comment]
Another error when tried from again after cloning


pkscreener/PKScreener/pkscreener/pkscreenercli.py", line 66, in <module>
    from PKDevTools.classes import log as log
ModuleNotFoundError: No module named 'PKDevTools'

**Reply by pkjmesra** [reply]
@vcrenjith - If you clone the repo, you will need to install ta-lib and the requirements. You can install ta-lib following steps as in https://github.com/pkjmesra/PKScreener/blob/main/.github/workflows/w1-workflow-build-matrix.yml and then install all requirements with `pip3 install -r requirements.txt`
